### PR TITLE
PDP Sharding documentation

### PR DESCRIPTION
**Pull request by omer9564**
new documentation available at /concepts/pdp-sharding/

### CODE REVIEW

**Comment by omer9564** [comment]
new documentation available [here](https://deploy-preview-128--permitio-docs.netlify.app/concepts/pdp-sharding/)

File: docs/concepts/pdp-sharding.mdx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                    address: $PDP-2-ADDRESS
+                    port_value: 80
+              metadata:
+                filter_metadata:
+                  envoy.lb:
+                    hash_key: 2
+    ```
+3. Run Envoy:
+    ```sh
+   docker run --rm -it --network=envoy \
```
Comment (roekatz) [inline docs/concepts/pdp-sharding.mdx:123]:
```suggestion
   docker run --rm -it --network= pdp-net \
```

File: docs/concepts/pdp-sharding.mdx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      -v $(pwd)/my-custom-envoy.yaml:/envoy-custom.yaml \
```
Comment (roekatz) [inline docs/concepts/pdp-sharding.mdx:124]:
```suggestion
      -v $(pwd)/pdp-lb-envoy.yaml:/envoy-custom.yaml \
```

File: docs/concepts/pdp-sharding.mdx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      envoyproxy/envoy:dev-d266941360e811eb34fc632d914211437e13f962 \
+          -c /envoy-custom.yaml
+    ```
+4. Run PDPs:
+    Each PDP should get its shard id as an environment variable:
+    ```sh
+    docker run -it \
+      -p 7766:7000 \
+      --name pdp-0 \
+      --network=envoy \
```
Comment (roekatz) [inline docs/concepts/pdp-sharding.mdx:136]:
```suggestion
      --network=pdp-net \
```

File: docs/concepts/pdp-sharding.mdx [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+      --env PDP_API_KEY=<YOUR_API_KEY> \
+      --env PDP_SHARD_ORD=0 \
+      permitio/pdp-v2:latest
+    ```
+    and
+    ```sh
+    docker run -it \
+      -p 7766:7000 \
+      --name pdp-1 \
+      --network=envoy \
```
Comment (roekatz) [inline docs/concepts/pdp-sharding.mdx:146]:
```suggestion
      --network=pdp-net \
```

File: docs/concepts/pdp-sharding.mdx [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+      permitio/pdp-v2:latest
+   ```
+
+Finally you can send an authorization check request to the Envoy container:
+
+<Tabs groupId="language">
+<TabItem value="curl" label="cURL">
+
+```bash
+curl http://localhost:10000/allowed \
```
Comment (roekatz) [inline docs/concepts/pdp-sharding.mdx:158]: Why is that needed?

File: static/img/diagrams/sharded-pdps-diagram.svg
Comment (roekatz) [inline static/img/diagrams/sharded-pdps-diagram.svg:1]: Looks good. Although you have redundant squares of PDP-2

**Review by roekatz — CHANGES_REQUESTED** [review]

File: docs/concepts/pdp-sharding.mdx (hunk 5 above)
Comment (omer9564) [reply docs/concepts/pdp-sharding.mdx:158]: to show the actual permit.check

File: static/img/diagrams/sharded-pdps-diagram.svg
Comment (omer9564) [reply static/img/diagrams/sharded-pdps-diagram.svg:1]: where ?

File: docs/concepts/pdp-sharding.mdx [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+The flow of an authorization request will be as follows:
+![Authorization Request](/img/diagrams/sharded-pdps-diagram.svg)
+
+
+To achieve sharding, multiple PDPs should be run.
+Currently, we assume a fixed number of PDPs, although scaling would be available soon.
+
+For the SDK’s queries to reach the correct PDP instance (holding the shard that has the correct tenant id), we are using Envoy.
+The SDK is configured with Envoy’s address as its PDP, and Envoy forwards the requests based on the Tenant ID to the corresponding PDP.
+Our backend sharding is implemented using that same algorithm, Thus both Envoy and our backend are mapping the same tenant ids to the same shard ids.
```
[the remark this answers is not in the document]
Comment (omer9564) [reply docs/concepts/pdp-sharding.mdx]: changed

File: docs/concepts/pdp-sharding.mdx [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+
+## Quickstart
+
+An Envoy container with a custom configuration should be deployed next to PDPs.
+See quick guide [here](https://www.envoyproxy.io/docs/envoy/latest/start/quick-start/run-envoy#override-the-default-configuration).
+The following configuration should be used:
+
+1. Create a docker network for the envoy and pdp containers to communicate:
+    ```sh
+    docker network create envoy
```
[the remark this answers is not in the document]
Comment (omer9564) [reply docs/concepts/pdp-sharding.mdx]: changed

File: docs/concepts/pdp-sharding.mdx (hunk 2 above)
Comment (omer9564) [reply docs/concepts/pdp-sharding.mdx]: changed

File: docs/concepts/pdp-sharding.mdx (hunk 3 above)
Comment (omer9564) [reply docs/concepts/pdp-sharding.mdx]: changed

File: docs/concepts/pdp-sharding.mdx (hunk 4 above)
Comment (omer9564) [reply docs/concepts/pdp-sharding.mdx]: changed

File: static/img/diagrams/sharded-pdps-diagram.svg
Comment (omer9564) [reply static/img/diagrams/sharded-pdps-diagram.svg:1]: changed it, please review again

File: docs/concepts/pdp-sharding.mdx (hunk 5 above)
Comment (roekatz) [reply docs/concepts/pdp-sharding.mdx:158]: Yes but I mean why the curl is needed? isn't the sdk enough?

**Review by roekatz — APPROVED** [review]
